software/controller/lib/debug/trace_cmd.cpp

```diff
@@ -29,17 +29,58 @@ ErrorCode TraceHandler::Process(Context *context) {
 
   // Sub-command kFlushTrace is used to flush the trace buffer
   // It also disables the trace
-  case static_cast<uint8_t>(Subcommand::kFlushTrace): {
+  case static_cast<uint8_t>(Subcommand::kFlush): {
```

Comment:
I think it would be cleaner to do something like
```
  auto subcmd = static_cast<Subcommand>(context->request[0]);
  switch (subcmd) {
```
at the top and avoid the casts in the switch statements. Might help the compiler detect overlooked cases as well if the right options are turned on.